libcudacxx/test/libcudacxx/std/numerics/c.math/fp_traits/fpclassify.pass.cpp

```diff
@@ -0,0 +1,128 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// <cmath>
+
+#include <cuda/std/cassert>
+#include <cuda/std/cmath>
+#include <cuda/std/limits>
+#include <cuda/std/type_traits>
+#include <cuda/type_traits>
+
+#include "test_macros.h"
+
+template <class T>
+__host__ __device__ void test_fpclassify(T val, int expected)
```

Comment:
One more thing, can you please check whether we can run some of those tests with the disabled conversions / constructors

We might need to add a header with generator functions but it would be awesome if we could get those tests to work with assuming functionality that might be user disabled